core/go.mod

```diff
@@ -4,22 +4,39 @@ go 1.20
 
 require (
 	github.com/cosmos/gogoproto v1.5.0
+	github.com/spf13/cobra v1.8.1
+	github.com/spf13/viper v1.19.0
```

Comment:
we should avoid adding these deps to core. Core is meant to be super minimal

Reply:
not sure how to avoid this though. But it cant live here. this increases the dependency graph significantly of core

Reply:
Easiest is to revert as it was and just have a store server component in server/v2. I'm not super a fan of seeing this in core either tbh (I do understand why we would need this if we wanted store/v2 to directly be a server component)

Reply:
There should be some abstraction, I don't think there is a need for things to depend on the raw implementation of store. Makes it harder to swap out components.

Reply:
I can abstract viper as an interface like:
`type AppOptions interface {
	Get(key string) any
}`

Another thing is `CLIConfig` struct just uses `Commands` filed (don't see usage of `Queries ` or `Txs `, otherwise they all `*cobra.Command`). Use `[]*cobra.Command` instead of `CLIConfig` should resolve this issue.

Reply:
> Another thing is CLIConfig struct just uses Commands filed (don't see usage of Queries or Txs , otherwise they all *cobra.Command). Use []*cobra.Command instead of CLIConfig should resolve this issue.

We should keep CLIConfig as it is. We want a server to be able to add commands under simd tx or simd q in addition to the root commands. This is quite useful for querying consensus specific stuff for instance, where for consistency it should be under query. We don't make use of all those fields now but that doesn't mean they aren't needed.

Reply:
Ah okay